controllers/v1/sprint_goal.go

```diff
@@ -0,0 +1,95 @@
+package v1
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	retrospectiveServices "github.com/iReflect/reflect-app/apps/retrospective/services"
+)
+
+// SprintController ...
+type SprintGoalController struct {
+	SprintService     retrospectiveServices.SprintService
+	PermissionService retrospectiveServices.PermissionService
+	TrailService      retrospectiveServices.TrailService
+}
+
+// Routes for Sprints
+func (ctrl SprintGoalController) Routes(r *gin.RouterGroup) {
+	r.POST("/", ctrl.Add)
+	r.GET("/", ctrl.List)
+	r.PUT("/:goalID/", ctrl.Update)
+	r.POST("/:goalID/", ctrl.Resolve)
+	r.DELETE("/:goalID/", ctrl.UnResolve)
```

Comment:
I don't think we should be using DELETE for marking a goal as unresolved/pending.

Reply:
changed